test/src/timeout_tests.erl

```diff
@@ -0,0 +1,131 @@
+                      control_action/4, control_action/5, control_action_opts/1,
+                      test_channel/0, expand_options/2, find_listener/0,
+                      info_action/4]).
+
+-include("rabbit.hrl").
+
+-define(TIMEOUT_LIST_OPS_PASS, 1000).
+
+all_tests() ->
+    passed = test_list_operations_timeout_pass(),
+    passed.
+
+test_list_operations_timeout_pass() ->
+    %% create a few things so there is some useful information to list
+    {_Writer, Limiter, Ch} = test_channel(),
+    [Q, Q2] = [Queue || Name <- [<<"foo">>, <<"bar">>],
+                        {new, Queue = #amqqueue{}} <-
+                            [rabbit_amqqueue:declare(
+                               rabbit_misc:r(<<"/">>, queue, Name),
+                               false, false, [], none)]],
+    ok = rabbit_amqqueue:basic_consume(
+           Q, true, Ch, Limiter, false, 0, <<"ctag">>, true, [], undefined),
+
+    %% list users
+    ok = control_action(add_user, ["foo", "bar"]),
+    {error, {user_already_exists, _}} =
+        control_action(add_user, ["foo", "bar"]),
+    ok = control_action(list_users, [], ?TIMEOUT_LIST_OPS_PASS),
+    
+    %% list parameters
+    ok = rabbit_runtime_parameters_test:register(),
+    ok = control_action(set_parameter, ["test", "good", "123"]),
+    ok = control_action(list_parameters, [], ?TIMEOUT_LIST_OPS_PASS),
+    ok = control_action(clear_parameter, ["test", "good"]),
+    rabbit_runtime_parameters_test:unregister(),
+
+    %% list vhosts
+    ok = control_action(add_vhost, ["/testhost"]),
+    {error, {vhost_already_exists, _}} =
+        control_action(add_vhost, ["/testhost"]),
+    ok = control_action(list_vhosts, [], ?TIMEOUT_LIST_OPS_PASS),
+
+    %% list permissions
+    ok = control_action(set_permissions, ["foo", ".*", ".*", ".*"],
+                        [{"-p", "/testhost"}]),
+    ok = control_action(list_permissions, [], [{"-p", "/testhost"}],
+                        ?TIMEOUT_LIST_OPS_PASS),
+
+    %% list user permissions
+    ok = control_action(list_user_permissions, ["foo"],
+                        ?TIMEOUT_LIST_OPS_PASS),
+
+    %% list policies
+    ok = control_action_opts(["set_policy", "name", ".*",
+                              "{\"ha-mode\":\"all\"}"]),
+    ok = control_action(list_policies, [], ?TIMEOUT_LIST_OPS_PASS),
+    ok = control_action(clear_policy, ["name"]),
+
+    %% list queues
+    ok = info_action(list_queues, rabbit_amqqueue:info_keys(), false,
+                     ?TIMEOUT_LIST_OPS_PASS),
+
+    %% list exchanges
+    ok = info_action(list_exchanges, rabbit_exchange:info_keys(), true,
+                     ?TIMEOUT_LIST_OPS_PASS),
+
+    %% list bindings
+    ok = info_action(list_bindings, rabbit_binding:info_keys(), true,
+                     ?TIMEOUT_LIST_OPS_PASS),
+
+    %% list connections
+    {H, P} = find_listener(),
+    {ok, C} = gen_tcp:connect(H, P, []),
+    gen_tcp:send(C, <<"AMQP", 0, 0, 9, 1>>),
+    timer:sleep(100),
```

Comment:
@videlalvaro I added a check for this, so it ensures that when `list_connections` is executed, a connection already exists. but what could cause the scenario you've mentioned to occur? i mean both the temp client node & rabbit node are started on the same host whenever the tests are run, and/or are likely to have through a number of amqp connection tests by the time they reach this point